File: 4-MedianOfTwoSortedArrays/4-MedianOfTwoSortedArrays.py
# Last updated: 09/10/2025, 09:40:33
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        nums1.extend(nums2)
        nums1.sort()
        print (nums1)
        if len(nums1) % 2 == 1:
            median =  len(nums1)//2
            return nums1[median]
        else:
            median = len(nums1)//2
            return (nums1[median]+nums1[median-1])/2.0
        return 0

File: 4-MedianOfTwoSortedArrays/4-MedianOfTwoSortedArrays.cpp
// Last updated: 03/03/2026, 20:07:45
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> nums;

        // Merge both arrays
        for (int x : nums1) nums.push_back(x);
        for (int x : nums2) nums.push_back(x);

        // Sort the merged array
        sort(nums.begin(), nums.end());

        int n = nums.size();

        // If even number of elements → average of middle two
        if (n % 2 == 0)
            return (nums[n/2 - 1] + nums[n/2]) / 2.0;

        // If odd number of elements → middle element
        return nums[n/2];
    }
};